src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/DocumentPullDiagnosticsEndpoint.cs

```diff
@@ -52,58 +57,71 @@ public TextDocumentIdentifier GetTextDocumentIdentifier(VSInternalDocumentDiagno
 
     public async Task<IEnumerable<VSInternalDiagnosticReport>?> HandleRequestAsync(VSInternalDocumentDiagnosticsParams request, RazorRequestContext context, CancellationToken cancellationToken)
     {
-        if (!_languageServerFeatureOptions.SingleServerSupport)
+        using (Track("diagnostics"))
```

Comment:
I would move this down, below the single server check. If we're not going to query the downstream servers, we don't need to track anything.